specification.markdown

```diff
@@ -228,6 +210,95 @@ VINT\_WIDTH | VINT\_MARKER | VINT\_DATA     | Element Data Size Status
         For example -2 can be coded as 0xFFFFFFFFFFFFFE or 0xFFFE or
         0xFE and 5 can be coded 0x000000000005 or 0x0005 or 0x05.
 
+## EBML Element Types
+
+Each defined EBML Element MUST have a declared Element Type. The Element Type defines a concept for storing data that may be constrained by length, endianness, and purpose.
+
+Element Data Type:   Signed Integer
+
+    Endianness:     Big-endian
+    Length:         A Signed Integer Element MUST at least 1 octet in
```

Comment:
Binary and master: not quite correct. For a binary a length of 0 may make sense in the theoretical case in which there's a default value for a binary element.

More important is the master case. This can actually happen in Matroska! Let's assume you have an audio track with a sampling frequency of 8000 Hz and one channel. In such a case the file must contain a TrackAudio master element, but that master may not contain any element as 8000 is the default value for SamplingFrequency and 1 is the default value for Channels.

This is exactly what happens if mkvmerge writes such a track as it sets libebml to skip elements set to their default values. Note that leaving out the TrackAudio element would NOT be the same thing semantically as elements with default values only apply if their parent element is present in the file.